### ISSUE TITLE
`--silent` flag is accepted but ignored when filtering

### ISSUE THREAD

**Issue by stabai** [pnpm/pnpm#4802]
When running a command from a subpackage directly (from within that folder), pnpm respects the `--silent` flag. But when running it from the root using `--filter`, the `--silent` flag is ignored.

The `--silent` flag is meant to suppress tooling output (like the lines that start with `>`) explaining what npm is doing.

### Code to reproduce the issue:

This can be reproduced with any subpackage script that logs to `stdout`.

Correctly suppresses tooling output:
```shell
cd foo/bar && pnpm --silent run baz
```

Does not suppress tooling output:
```shell
pnpm --silent --filter foo-bar run baz
```

### Expected behavior:
```
OUTPUT GOES HERE
```

### Actual behavior:
```

> foo-bar@0.0.1 baz /Users/stabai/code/foo/bar
> node baz.js

OUTPUT GOES HERE
```

### Additional information:

 - `node -v` prints: v17.6.0
 - Windows, macOS, or Linux?: macOS

**Comment by stabai** [comment]
I created a shared environment reproducing the issue:
https://replit.com/@ShawnTabai/pnpm-reproduce-issue-4802#repro.sh

**Comment by leondreamed** [comment]
Is there any known workaround for this issue?

**Comment by stabai** [comment]
I haven't found one 😔

**Comment by mshick** [comment]
You could try `pnpm --filter my_filter exec -- pnpm --silent my-script`

**Comment by arxpoetica** [comment]
@mshick that worked. Would be nice for this bug to be fixed, though.

**Comment by AtaTaheri** [comment]
version 8.9.0 here and still exists

**Comment by damianobarbati** [comment]
any news on this guys?

**Comment by divmgl** [comment]
Yep, also having this issue. Just ran into a serious script bug because of it.

**Comment by stefanhalbritterpdg** [comment]
I wanted to run a script from `package.json` in the pipeline
The task of the typescript file was to "echo" a token to export it to an environment variable

Due to the failing silent mode on `--filter` for my sub package, the output was polluted with tooling logs. I had to invoke the script directly with `node` as a workaround

**Comment by pleunv** [comment]
This appears to have been resolved in v9 so this issue can be closed.

**Comment by zkochan** [comment]
This no longer reproduces on current pnpm, so closing.

Verified on pnpm 12.5.1: `pnpm --silent --filter <project> run build` prints only the script's output, the same as running it inside the project.

If you still hit this on the latest pnpm, please comment with a reproduction and we will reopen.

---
Written by an agent (Claude Code, claude-opus-5).
